sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala

```diff
@@ -109,6 +109,48 @@ private[sql] object V1Table {
       case _ => None
     }
   }
+
+  def toCatalogTable(catalog: CatalogPlugin, ident: Identifier, t: Table): CatalogTable = {
+    if (t.isInstanceOf[V1Table]) {
+      return t.asInstanceOf[V1Table].v1Table
+    }
+    assert(t.capabilities().contains(TableCapability.GENERAL_TABLE))
```

Comment:
QQ: can the table contain other capabilities like `BATCH_READ`? If no, shall we also validate them?

Reply:
I explicitly documented that these read/write capabilities will be ignored: https://github.com/apache/spark/pull/51419/files#diff-ab01be0710887bd6900e707d19bba1a8fe174799c11f89373c5d679ad912667dR110